src/crimson/net/ProtocolV1.cc

```diff
@@ -595,6 +595,45 @@ void ProtocolV1::start_accept(SocketFRef&& sock,
 
 // open state
 
+seastar::future<> ProtocolV1::write_messages(std::queue<MessageRef>& msgs)
+{
+  bufferlist bl;
```

Comment:
You might want to `::reserve()` required space for the C-string taking `::append()` calls:
```cpp
  bl.reserve(msgs.size() * (sizeof(CEPH_MSGR_TAG_MSG) + sizeof(header) + sizeof(footer)));
```

This should prevent from allocating 4k at first call and wasting memory when the queue has few messages.

Reply:
Sorry not familiar with bufferlist interface, can I still use `bl.append(...)` to fill the C-strings, after calling `bl.reserve(...)`?

Reply:
Yup, it's OK.